src/site/content/en/learn/pwa/service-workers/index.md

```diff
@@ -1,11 +1,137 @@
+
+{% Aside 'gotchas' %}
+Not all browsers support service workers, and your service worker won't be available on first load or while it's waiting to activate. Therefore, you should treat it as optional and not require it for core functionality of your PWA.
+{% endAside %}
+
+## Registering a service worker
+
+Before a service worker takes control of your page, it  must be registered for your PWA. That means the first time a user comes to your PWA, network requests will go directly to your server because the service worker is not yet in control of your pages.
+
+After checking if the browser supports the Service Workers API, your PWA can register a service worker. When loaded, the service worker sets up shop between your PWA and the network, intercepting requests and serving the corresponding responses.
+
+```js
+if ('serviceWorker' in navigator) {
+   navigator.serviceWorker.register("/serviceworker.js");
+}
+```
+
+{% Glitch 'learn-pwa-service-worker-registration' %}
+
+{% Aside %}
+There is only one service worker per PWA, but that doesn't mean you need to place the code only in one file. A service worker can include other files using [`importScripts`](https://developer.mozilla.org/docs/Web/API/WorkerGlobalScope/importScripts) in every browser or using [ECMAScript module imports](/es-modules-in-sw/) in some modern browsers.
+{% endAside %}
+
+### Verify if a service worker is registered
+
+To verify if a service worker is registered, use developer tools in your favorite browser.
+
+Using DevTools on Chromium-based browsers, such as Microsoft Edge, Google Chrome, or Samsung Internet, and also on Firefox:
+
+1. In the **Application** category, in the left pane, select **Service Workers**
+1. Check that the service worker's script URL appears with the status "Activated"(You'll learn what this status means in the lifecycle section in this chapter). On Firefox the status that is shown could be "Running" or "Stopped"
+
+In Safari:
+
+1. Open the **Develop** menu
+1. Open the **Service Workers** submenu
+1. Check that an entry in the submenu appears with your current origin that opens an inspector over the service worker's context
+
+<figure>
+{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/QMsYbKeQ6OJSm0FeqFxG.png", alt="Service Worker developer tools on Chrome, Firefox and Safari.", width="800", height="296" %}
+  <figcaption>Service Worker developer tools on Chrome, Firefox and Safari.</figcaption>
+</figure>
+
+{% Aside %}
+You can remotely inspect a phone or tablet from your desktop browser when you want to check service worker registration status over a mobile device. You will see details on how to do that in the [Tools and Debug chapter](/tools-and-debug)
+{% endAside %}
```

Comment:
This is broader than the existing Chrome information I believe, though it's not written yet. However I'll ensure it links to rather than duplicates content. I think there is place for different styles of learning (otherwise there would only ever be one article on a subject).